internal/cmd/get/get.go

```diff
@@ -45,128 +42,25 @@ func NewCommand(ctx context.CLIContext) *cobra.Command {
 }
 
 func run(ctx context.CLIContext, opts *options, _ *cobra.Command, args []string) {
-	// Find directory for installing packages
-	var projectConfig *schema.ProjectConfig
-	var err error
-	var scope dependency.ScopeType
-	var installedDeps []schema.Dependency
-
-	depsMgr := dependency.NewManager(ctx)
-	depsMgr.DefaultRegistry = ctx.Config.DefaultRegistry
+	var getScope scope.Scope
 
 	if opts.Global {
-		scope = dependency.GlobalScope
-		if ctx.Paths.GlobalInstallDir == "" {
-			log.Fatal("Cannot init global install directory")
-		}
-	} else {
-		scope = dependency.ProjectScope
-
-		if !opts.NoInit && ctx.Paths.ProjectInstallDir == "" {
-			ctx, err = initialiseProjectInCurrentDir(ctx)
-			if err != nil {
-				log.Fatalf("Failed to initialise project in the current dir: %s", err)
-			}
-			depsMgr = dependency.NewManager(ctx)
-		}
-
-		if ctx.Paths.ProjectInstallDir == "" {
-			log.Fatal(`Packages can be installed locally only under project directory, use "--global"`)
-		}
-		projectConfig, err = schema.LoadProjectConfig(ctx.Paths.ProjectConfigFile)
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		if projectConfig.DefaultRegistry != "" {
-			depsMgr.DefaultRegistry = projectConfig.DefaultRegistry
-		}
-	}
-
-	var toInstall []schema.Dependency
-
-	if len(args) == 0 && !opts.Global {
-		toInstall = projectConfig.Dependencies
+		getScope = scope.NewGlobal(ctx.Paths.GlobalInstallDir, opts.From, opts.As, opts.Version)
```

Comment:
Options *from*, *as* and *version* are describing command which is going to be installed, not scope. These options apply also to commands installed locally.